ext/AbstractDifferentiationChainRulesCoreExt.jl

```diff
@@ -3,11 +3,22 @@ module AbstractDifferentiationChainRulesCoreExt
 import AbstractDifferentiation as AD
 using ChainRulesCore: ChainRulesCore
 
-AD.@primitive function pullback_function(ba::AD.ReverseRuleConfigBackend, f, xs...)
-    _, back = ChainRulesCore.rrule_via_ad(AD.ruleconfig(ba), f, xs...)
-    pullback(vs) = Base.tail(back(vs))
-    pullback(vs::Tuple{Any}) = Base.tail(back(first(vs)))
-    return pullback
+AD.@primitive function value_and_pullback_function(ba::AD.ReverseRuleConfigBackend, f, xs...)
+    value, back = ChainRulesCore.rrule_via_ad(AD.ruleconfig(ba), f, xs...)
+    function value_and_pullback(vs)
+        pb_value = if vs === nothing
```

Comment:
We have to handle `nothing` due to the way how `jacobian` is implemented for `@primitive function value_and_pullback_function` (this PR) and `@primitive function pullback_function` (on the master branch): https://github.com/JuliaDiff/AbstractDifferentiation.jl/blob/041d760d4c0bff8a5e023f9d1d5cad2ffbb68f4d/src/AbstractDifferentiation.jl#L551

Reply:
On the master branch, the default fallback for `value_and_pullback_function` handles `nothing` inputs, hence the design choice becomes apparent only in this PR.

Reply:
I'm not sure why the `jacobian` is implemented in this way. In principle, calling `pullback` repeatedly should be sufficient and the `value` should not be needed.

Reply:
Could be solved easily with https://github.com/JuliaDiff/AbstractDifferentiation.jl/pull/35 I assume.